### ISSUE TITLE
Hexapdf Document not opened in Adobe Acrobat Reader when I added print permission

### ISSUE THREAD

**Issue by ramir2426** [gettalong/hexapdf#71]
hi @gettalong !!
I'm using hexapdf to generate pdf document but it's not opening in adobe reader when I added a print permission in the document.
the error message.
`there was an error opening this document. this file is damaged and could not be repaired`
if I removed this permission then document opened in Adobe Acrobat Reader.
Could you help me with that problem?
thanks and regards.

**Comment by gettalong** [comment]
First: Please avoid such sentences like "please gimme a solution to tackle this problem." when you are participating on this issue tracker; they come across as demanding. You could rephrase it to "Could you help me with that problem?" - thanks!

Can you provide the code to generate such a faulty PDF or the generated, faulty PDF in question? Otherwise it is very hard to debug.

**Comment by ramir2426** [comment]
@gettalong first of all I want to thanks for correcting myself.
here is the code.

    HexaPDF::Document.open(file_path) do |doc|
        doc.encrypt(permissions: [:print])
        doc.write('example.pdf')
    end

the file which is located in  this `file_path`  path is also generated with hexapdf and this file open correctly in Adobe Acrobat Reader but when I added the permission with help of above code then file is not opened and error occurring.

**Comment by gettalong** [comment]
@ramir2426 Thank you for the additional information - I was successful in reproducing the problem and will investigate it!

**Comment by gettalong** [comment]
@ramir2426 I'm still trying to figure out why there are some cases that are not working in Adobe Reader... They open fine in every other viewer... What's more is that the encryption works fine if used without object streams but fails with them...

**Comment by genericsteele** [comment]
@gettalong I'm following along with this issue. What would be the consequences of optimizing without object streams?

**Comment by gettalong** [comment]
@genericsteele @ramir2426 I think I already found the problem and it seems to be a bug in Adobe Acrobat... Will test some more and release a bug fix version.

@genericsteele As for you question: Object streams compress non-stream objects with Deflate, so if the file has many objects, object streams provide quite substantial space savings.

Also see https://gist.github.com/gettalong/7602fe9e9b0b71fd094e9f6b815fafbd

**Comment by gettalong** [comment]
@ramir2426 @genericsteele The work around for this Acrobat bug is available with the new 0.9.3 releae.

**Comment by genericsteele** [comment]
@gettalong Fantastic work. I just tested it and it works beautifully

**Comment by gettalong** [comment]
@genericsteele Great, thank you!

**Comment by faizaankhan** [comment]
@gettalong This seems to repeat again. Checked on the latest 0.23. Merged PDFs are throwing errors.

**Comment by gettalong** [comment]
@faizaankhan Can you provide a script to reproduce the error? A file created with the script in https://github.com/gettalong/hexapdf/issues/71#issuecomment-496100756 works fine in Adobe Reader when run on 0.23
